test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSqlCompatibilityTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema;
+using Microsoft.Health.Fhir.Tests.Common;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.Integration.Persistence
+{
+    public class SqlServerSqlCompatibilityTests
+    {
+        /// <summary>
+        /// A basic smoke test verifying that the code is compatible with schema versions
+        /// all the way back to <see cref="SchemaVersionConstants.Min"/>. Ensures that the code can
+        /// insert insert a resource using every schema version, and that we can read resources
+        /// that were inserted into with earlier schemas.
+        /// </summary>
+        [Fact]
+        public async Task GivenADatabaseWithAnEarlierSupportedSchema_WhenUpserting_OperationSucceeds()
```

Comment:
+1 this test is awesome, thanks for adding it.

My understanding is, if the current schema version is `x` and version `x+1` is added, this would test that `x+1.diff.sql` can be applied on top of schema version `x`. 

Would it also be possible to test the scenario when the snapshot file, `x+1.sql`, is run? I believe this has a different initialization path in `SqlServerFhirModel.cs` that would be good to verify (np if we want to make a work item to add this in later).

Reply:
Dug up [AB#78858](https://microsofthealth.visualstudio.com/Health/_workitems/edit/78858/) which we could use to track this?

Reply:
Ok, yes, I see. We can expand the test to also ensure we create the database from each snapshot version, and then upgrade from there. Is that what you had in mind?

Reply:
Yep, I think even simply creating the database from each snapshot version and posting a resource would be good, since we are already testing the upgrade paths in this test.

Reply:
Added in https://github.com/microsoft/fhir-server/pull/1606/commits/c09b5d12d283143adcb7981f5f7797479fa7f72e